jdat_notebooks/ifu_optimal/ifu_optimal.ipynb

```diff
@@ -0,0 +1,821 @@
+    "    hdr=hdulist[1].header\n",
+    "    hdulist.info()\n",
+    "    print(w)\n",
+    "    \n",
+    "#Load with Spectrum1D    \n",
+    "spec1d = Spectrum1D.read(filename)\n",
+    "\n",
+    "#Wavelengths\n",
+    "wavelength=np.array(spec1d.spectral_axis.value)\n",
+    "print(wavelength)\n",
+    "\n",
+    "#Sum over spaxels\n",
+    "fnu_sum=np.sum(spec1d.flux,axis=(0,1))\n",
+    "\n",
+    "#List of cube slices\n",
+    "data=[]\n",
+    "var=[]\n",
+    "spec1d_len=len(spec1d.spectral_axis.value)\n",
+    "for idx in range(spec1d_len): \n",
+    "    data.append(sci[idx,:,:])\n",
+    "    var.append(err[idx,:,:])   #variance=err, not variance=err**2.  Squaring the err gives strange, noisy results. \n",
+    "\n",
+    "#Window data, variance, and PSF model (and replace NaNs)\n",
+    "#The existing JWST pipeline window is overgenerous (39x33 instead of the nominal 30x30 pixels)\n",
+    "data_win=np.nan_to_num(np.array(data)[:,5:-4,3:])\n",
+    "data_var=np.nan_to_num(np.array(var)[:,5:-4,3:]) \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:* Can we fix or suppress this warning?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:*  Is there a way to read in with only Spectrum1D to perform all of our cube operations, not using fits.open()?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize Science Data with Cubeviz"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Note:* Cubeviz incompatible with jupyter_client 6.1.6.  Use jupyter_client 5.3.5 instead."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from jdaviz import CubeViz\n",
+    "cubeviz = CubeViz()\n",
+    "cubeviz.app"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### UI Instructions:\n",
+    "* Load science datacube into Cubeviz using the cell below\n",
+    "* Select a circular subset region centered on the source. \n",
+    "* Note that the region is pixelated and doesn't include fractional pixels\n",
+    "* Change the collapse method to 'Sum' in spectrum viewer: Gear icon : Viewer \n",
+    "* --This 'Sum' method yields our subset extraction\n",
+    "* Change the vertical zoom to see the spectral features in the Subset spectrum"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Developer Notes*: \n",
+    "(1) Image viewer contrast settings change when you click on the side bar to expand/contract jupyter scroll window\n",
+    "(2) Spectrum viewer: Viewer cube collapse method should default to Sum (not Maximum)\n",
+    "(3) Spectrum viewer y scale returns to autoscal3 when the region is moved, and y-zoom has to be adjusted again"
+   ]
```

Comment:
I did not find these notes helpful. I was unable to change the contrast in the viewer.

Reply:
These 3 notes are Developer Notes, not notes to the User.